solutions/fully-configurable/main.tf

```diff
@@ -32,3 +39,121 @@ module "scc_wp" {
 }
+
+########################################################################################################################
+# Cloud Security Posture Management (CSPM)
+########################################################################################################################
+
+resource "ibm_resource_instance" "app_configuration_instance" {
+  plan              = "basic"
+  name              = "${var.prefix}-conf-agg"
+  location          = var.region
+  resource_group_id = module.resource_group.resource_group_id
+  service           = "apprapp"
+}
+
+# Trusted Profile for Workload Protection
+resource "ibm_iam_trusted_profile" "workload_protection_profile" {
+  count = var.cspm_enabled ? 1 : 0
+  name  = "${var.prefix}-workload-protection-trusted-profile"
+}
+
+# Null resource to enable CSPM via CLI
+resource "null_resource" "enable_cspm" {
+  count = var.cspm_enabled ? 1 : 0
+
+  provisioner "local-exec" {
+    command = <<EOT
+      ibmcloud login --apikey ${var.ibmcloud_api_key} -g ${module.resource_group.resource_group_name} --no-region && \
+      ibmcloud resource service-instance-update ${module.scc_wp.id} \
+        -p '{"enable_cspm": true, "target_accounts": [{"account_id": "${data.ibm_iam_account_settings.iam_account_settings.account_id}", "config_crn": "${ibm_resource_instance.app_configuration_instance.crn}", "trusted_profile_id": "${ibm_iam_trusted_profile.workload_protection_profile[0].id}"}]}' \
+        -g ${module.resource_group.resource_group_name}
+    EOT
+  }
+
+  depends_on = [
+    module.scc_wp,
+    ibm_iam_trusted_profile.workload_protection_profile
+  ]
+}
+
+# Trusted Profile Policiy for All Identify and Access enabled services for WP
+resource "ibm_iam_trusted_profile_policy" "policy_workload_protection_apprapp" {
+  count       = var.cspm_enabled ? 1 : 0
+  profile_id  = ibm_iam_trusted_profile.workload_protection_profile[0].id
+  roles       = ["Service Configuration Reader", "Viewer", "Configuration Aggregator Reader"]
+  description = "apprapp"
+  resources {
+    service = "apprapp"
+  }
+}
+
+# Trusted Profile Trust Relationship for Config Service
+resource "ibm_iam_trusted_profile_identity" "trust_relationship_workload_protection" {
+  count         = var.cspm_enabled ? 1 : 0
+  identifier    = module.scc_wp.crn
+  identity_type = "crn"
+  profile_id    = ibm_iam_trusted_profile.workload_protection_profile[0].id
+  type          = "crn"
+}
+
+##############################################################
+# App Config
+##############################################################
+
+module "crn_parser" {
+  source  = "terraform-ibm-modules/common-utilities/ibm//modules/crn-parser"
+  version = "1.1.0"
+  crn     = ibm_resource_instance.app_configuration_instance.crn
+}
+
+# Trusted Profile for Config Service
+resource "ibm_iam_trusted_profile" "config_service_profile" {
+  count = var.cspm_enabled ? 1 : 0
+  name  = "${var.prefix}-config-service-trusted-profile"
```

Comment:
prefix is optional, so you need to change this logic to handle appropriately. Also you need to expose the trusted profile name as a variable too (it can default to `config-service-trusted-profile`)